### PR TITLE
Update Button component to support outputting standalone CSS

### PR DESCRIPTION
**Pull request by sghoweri**
^ for externally using Bolt as a standalone library via <link> and <script> tags.

For example, allowing our teams to still get some of the benefits of having a Design System even when full on Drupal integration isn't currently possible for a variety of reasons (however dropping in a link tag on the page IS an option):
```
<link rel="stylesheet" href="https://unpkg.com/@bolt/components-button@0.7.0/dist/bolt-button.min.css">
```

This update also includes minor cleanup of the existing button component to remove out of date demos that need to get redone in order for these examples to stay synced up with the rest of the system. Clearing these out to make way for new work on these.

### CODE REVIEW

**Comment by sghoweri** [comment]
@remydenton @mikemai2awesome can either of you guys take a quick peek at this? 

I just fixed the merge conflicts -- just need this small update to get OK'd so we can get this version of the button component down to master.

File: build-tools/gulp-tasks/build-webpack/webpack.config.js [hunk 1]
```diff
@@ -34,6 +34,7 @@ const defaultConfig = {
   entry: {
     'critical-fonts': './src/_patterns/02-components/bolt-critical-fonts/src/critical-fonts',
 
+    'critical-fonts': './src/_patterns/02-components/bolt-button/src/button',
```
Comment (remydenton) [inline build-tools/gulp-tasks/build-webpack/webpack.config.js]: This isn't right, is it?  Two entries for `critical-fonts`?

Comment (remydenton) [reply build-tools/gulp-tasks/build-webpack/webpack.config.js]: Never mind, I see you removed it in a later commit.

**Comment by remydenton** [comment]
This passes a quick gut check.  I will try to take a closer look a little later (feel free to merge it in in the meantime if you're trying to get this out the door).

**Pull request merged**